### DISCUSSION TITLE
Heads up about 3.12.4

### DISCUSSION THREAD

**Discussion by paul-stork** [reflex-dev/reflex#3469] [General]
Good day, good people.

I spent the day trying to figure out a bug in my Docker container all day that I just couldn't wrap my brain around:

TypeError: ForwardRef._evaluate() missing 1 required keyword-only argument: 'recursive_guard'

I couldn't figure it out because nothing in my CODE had changed. However...

Because I was pulling Docker image python:3.12-slim-bookworm as my base image, I missed that 3.12.4 had been pushed as the base image. This change contains this problem:

Fix missing recursive_guard parameter in Pydantic v1 for python 3.12.4+ #9611
https://github.com/pydantic/pydantic/pull/9611

After setting the base image python:3.12.3-slim-bookworm it was working again.

So, long story short: Avoid 3.12.4 for the time being.

**Comment by masenf** [comment]
Thanks for raising this issue. Looks like pydantic has already merged the fix, so we're waiting for a release from them which will work with 3.12.4.

There might be some changes in the reflex pyproject.toml to ensure that newer interpreter versions have the fixed pydantic package set as their minimum.

**Reply by benedikt-bartscher** [reply]
fixed in [pydantic v2.7.4](https://github.com/pydantic/pydantic/releases/tag/v2.7.4)
